aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -239,6 +240,159 @@ Tensor wrapped_scalar_tensor(
+// 3. Create a mask for locations of adjacent duplicate values within the
+//    sorted list. Duplicate values are in both elements and test elements.
+// 4. Reorder the mask to match the pre-sorted element order.
+// 5. Index the mask to match the pre-unique element order. If
+//    assume_unique is true, just take the first N items of the mask,
+//    where N is the original number of elements.
+static void isin_sorting(const Tensor& elements, const Tensor& test_elements,
+    bool assume_unique, bool invert, Tensor& out) {
+  Tensor elements_flat, test_elements_flat, unique_order;
+  if (assume_unique) {
+    elements_flat = elements.ravel();
+    test_elements_flat = test_elements.ravel();
+  } else {
+    std::tie (elements_flat, unique_order) = at::_unique(
+        elements, /*sorted=*/ false, /*return_inverse=*/ true);
+    std::tie (test_elements_flat, std::ignore) = at::_unique(test_elements, /*sorted=*/ false);
+  }
+
+  Tensor all_elements = at::_cat({elements_flat, test_elements_flat});
+  Tensor sorted_elements, sorted_order;
+  std::tie (sorted_elements, sorted_order) = all_elements.sort(
+      /*stable=*/ true, /*dim=*/ 0, /*descending=*/ false);
+
+  Tensor duplicate_mask = at::empty(all_elements.sizes(),
+    TensorOptions(elements.device()).dtype(ScalarType::Bool));
+  auto iter = TensorIteratorConfig()
+    .set_check_mem_overlap(true)
+    .add_output(duplicate_mask)
+    .add_input(sorted_elements)
+    .cast_common_dtype_to_outputs(false)
+    .enforce_safe_casting_to_output(false)
+    .check_all_same_dtype(false)
+    .build();
+  AT_DISPATCH_INTEGRAL_TYPES(elements.scalar_type(), "isin_sorting", [&]() {
+    iter.for_each([&](char** data, const int64_t* strides, int64_t n) {
+      auto* mask_bytes = data[0];
+      const auto* element_bytes = data[1];
+
+      for (int i = 0; i < n - 1; ++i) {
+        auto* mask_data = reinterpret_cast<bool*>(mask_bytes);
+        const scalar_t element_val = *(reinterpret_cast<const scalar_t*>(element_bytes));
+        const scalar_t next_element_val = *(reinterpret_cast<const scalar_t*>(element_bytes + strides[1]));
+        *mask_data = invert ? (element_val != next_element_val) : (element_val == next_element_val);
+
+        mask_bytes += strides[0];
+        element_bytes += strides[1];
+      }
+
+      // Last duplicate mask item is unconditionally true or false based on invert flag.
+      auto* mask_data = reinterpret_cast<bool*>(mask_bytes);
+      *mask_data = invert;
+    });
+  });
+
+  Tensor mask = at::empty(all_elements.sizes(), TensorOptions(elements.device()).dtype(ScalarType::Bool));
+  mask.index_copy_(0, sorted_order, duplicate_mask);
+
+  if (assume_unique) {
+    out.copy_(mask.slice(0, 0, elements.numel()).view_as(out));
+  } else {
+    out.copy_(at::index(mask, {c10::optional<Tensor>(unique_order)}));
+  }
+}
+
+Tensor& isin_out(const Tensor& elements, const Tensor& test_elements, bool assume_unique, bool invert, Tensor& out) {
+  out.resize_as_(elements);
```

Comment:
PyTorch currently does resize out tensors to be the appropriate size, but we're deprecating this behavior. Use resize_output() to ensure the out tensor is the correct size. See:

https://github.com/pytorch/pytorch/blob/e29d8477a61d90063cb9ef0c57a735ff6fbd2b5f/aten/src/ATen/native/Resize.h#L19